### ISSUE TITLE
enable-termination-protection flag

### ISSUE THREAD

**Issue by vrangasam** [awslabs/aws-sam-cli#1752]
**Description:**

CloudFormation's `create-stack` offers `enable-termination-protection` flag to protect a stack from getting deleted.

Is it possible to offer that flag in `sam`?

**Comment by aahung** [comment]
some info about possible implementation.

SAM CLI uses `create-change-set` API to create stacks, which does not have `enable-termination-protection` flag as `create-stack` does. If we choose to add this flag, we might need to call `update-termination-protection` API after creation

**Comment by yuri1969** [comment]
Still using a workaround chaining the `sam` calls with `aws cloudformation update-termination-protection --enable-termination-protection --stack-name "$STACK_NAME"`.

**Comment by AVandana-1806** [comment]
Any plan on adding this feature to sam cli?

**Comment by valerena** [comment]
One additional comment: The difference here is that CloudFormation has different operations for Create and Update. `create-stack` has `--enable-termination-protection`, but `update-stack` doesn't. The idea behind it is that this is something that you set initially in a stack and it shouldn't change much. In SAM CLI, both operations happen under `sam deploy`, so it's hard to identify the user's intent. 

It's still a reasonable feature, but I just wanted to emphasize a little bit on why is this not as straightforward as it might look like.

**Comment by AVandana-1806** [comment]
@valerena Yes that is right but we don't have to do it a part of sam deploy, We can create a new SAM command which can enable and disable the termination.
